app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarUIView.kt

```diff
@@ -87,14 +93,63 @@ class ToolbarUIView(
     }
 
     override fun updateView() = Consumer<SearchState> {
-        if (it.isEditing) {
-            view.url = it.query
+        if (shouldUpdateEngineIcon(it)) {
```

Comment:
I personally like how clean/clear this feels, but let me know if you think I went overboard on the abstraction in this function!

Reply:
No, this looks good to me.